backend/spindle/views.py

```diff
@@ -7,51 +8,81 @@
 import base64
 
 
+class SpindleError(Enum):
+    SPINDLE = "spindle"
+    LATEX = "latex"
+    GENERAL = "general"
+
+
+class SpindleResponse:
+    def __init__(
+        self,
+        tex: Optional[str] = None,
+        pdf: Optional[str] = None,
+        redirect: Optional[str] = None,
+        error: SpindleError = None,
+    ):
+        self.tex = tex
+        self.pdf = pdf
+        self.error = error
+        self.redirect = redirect
+
+    def jsonResponse(self) -> JsonResponse:
+        return JsonResponse(
+            {
+                "tex": self.tex,
+                "pdf": self.pdf,
+                "redirect": self.redirect,
+                "error": self.error.value if self.error else None,
+            }
+        )
+
+
 class SpindleView(View):
     def post(self, request: HttpRequest, mode: Literal["tex", "pdf", "overleaf"]):
         user_prompt = request.body.decode("utf-8")
 
         # Sending data to Spindle container.
-        response = urllib3.request(
-            method="POST", url="http://localhost:32768/", body=user_prompt
+        spindle_reponse = urllib3.request(
+            method="POST",
+            url="http://localhost:32768/",
+            body=user_prompt,
+            headers={"Content-Type": "application/json"},
```

Comment:
The request from the frontend is here directly propagated to the Spindle container. I would at least add some lines of validation; somebody who reads the code will then also understand what the format of the data is.

By the way, I think it would be a good idea to use Django Rest Framework for the API views, which has JSON support built in. (But for some reason there is no stable version of DRF yet for Django 4.2...)

Reply:
Done! I will put the conversion to DRF for a separate (small) issue, but I do think it is a good idea.